src/Listener.php

```diff
@@ -64,10 +64,48 @@ public function setTagNameCase($case)
      *                                 It uses array_walk() internally.
      *                                 The suggested signature for the argument is:
      *                                 function (string &$value, string $tag);
+     *
+     * @deprecated
      */
     public function setTagValueProcessor(callable $processor = null)
     {
-        $this->tagValueProcessor = $processor;
+        if (is_null($processor)) {
+            $this->tagValueProcessors = [];
+
+            return;
+        } // else
+        $this->tagValueProcessors = [$processor];
+    }
+
+    /**
+     * @param  $processor Function or array of functions to be applied to every member
+     *                    of an BibTeX entry. Uses array_walk() internally.
+     *                    The suggested signature for each function argument is:
+     *                        function (string &$value, string $tag);
+     *                    Note that functions will be applied in the same order
+     *                    in which they were added.
+     * @throws \InvalidArgumentException
+     */
+    public function addTagValueProcessor($processor)
+    {
+        if (is_array($processor)) {
```

Comment:
I think the right approach here is to invert the order of the "if"s in the code. Then, if the `$processor` passed is a callable array, it will answer `true` to `is_callable`, and will be processed adequately. If the array is not callable, it will proceed to the next step, and process each item. Does this make sense?

I'm writing the tests and will submit them soon.